stl/src/filesystem_space.cpp

```diff
@@ -42,12 +42,11 @@ _EXTERN_C
         _Last_error = __std_win_error{GetLastError()};
     } else {
         {
-            constexpr DWORD _Static_size = MAX_PATH;
-            wchar_t _Temp_buf[_Static_size];
-            _Last_error = _Fs_space_attempt(_Temp_buf, _Static_size, _Target, _Available, _Total_bytes, _Free_bytes);
-            if (_Last_error == __std_win_error::_Success) {
-                return __std_win_error::_Success;
+            if (GetDiskFreeSpaceExW(_Target, reinterpret_cast<PULARGE_INTEGER>(_Available), 

```

Comment:
The [documentation](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getdiskfreespaceexw) says the first argument to `GetDiskFreeSpaceEx` is "A directory on the disk" (presumably a pathname of such a directory) whereas [fs.op.space]/1 says `filesystem::space` behaves like [POSIX `statvfs` which accepts a pathname of any file on a disk](https://pubs.opengroup.org/onlinepubs/009695399/functions/statvfs.html). Is the documentation of `GetDiskFreeSpaceEx` incorrect, or do we need to fix this problem?

Reply:
Seems like we should add a test for it.

Reply:
GetDiskFreeSpaceEx definitely needs a directory. This is what GetVolumePathNameW gave us, though it did have that extra permissions requirement. Could we use parent_directory if we're dealing with a file?

Reply:
I added a test for it.